### DISCUSSION TITLE
Follow the development flow: design, deploy, runtime

### DISCUSSION THREAD

**Discussion by lucascanna** [mia-platform/community#203] [Ideas]
Hi all,

My classical usage of the Mia-Platform Console is:
* do things on the Design area and save the configuration
* deploy what I have just configured
* check the Runtime to see if everything is okay

It would be great if Mia-Platform Console would guide me in this flow by adding the following:
* A button that let me jump to the Deploy section right after saving the configuration
* A button that let me jump in the Runtime section right after the deploy 

What do you guys think about it?

Have a nice day!

**Comment by fredmaggiowski** [comment]
Hey @Lucascanna 

In v10.4.0 the Deploy section has received some minor improvements and it shows a button to go easily in the runtime section, do you wish for something more?

<img width="694" alt="image" src="https://user-images.githubusercontent.com/7142570/221916133-61db0286-2b48-4e5b-99e9-67f6086568e4.png">

As for the hint to go to the deploy section after saving the configuration, that's something that has been implemented but has not been activated yet since we want to improve it a little bit before releasing it in production. @albertotessarotto we might prioritize it a little bit.

**Reply by davidebianchi** [reply]
Yes, for the `go to deploy` UX there are a lot of workflow where the user save more time on the design area before deploy. So, the navigation to the deploy section should not be invasive to the other common workflows.

**Reply by albertotessarotto** [reply]
Sure, in considering Design -> Deploy flow, we were also taking into account not becoming invasive precisely because of the use case mentioned by Davide. We will try to raise the priority of it and find the most smart way for the solution.

**Reply by lucascanna** [reply]
Hi guys! Thanks for your replies. I am happy to hear that you are considering to improve this flow.

I only have one feedback with regards to the new feature mentioned by @fredmaggiowski.
I didn't notice this feature because usually I don't wait for the pipeline to finish. I jump in the Runtime section as soon as the pipeline has started. Therefore, I would like to have that button already in the modal of the running pipeline.

Moreover, sometimes the user is uncertain about whether he can close that modal before the pipeline has finished to run. The "Monitor Runtime status" button in that modal would remove all doubts to the user.

**Reply by albertotessarotto** [reply]
Nice consideration, I share your feedback even with @jacopostefani and @Danielecina. It is worth investigating whether this might be a common habit among our users to implement the suggested improvement.

**Reply by fredmaggiowski** [reply]
@jacoposte FYI the tag in the previous message didn't work

**Comment by akelity** [comment]
This feature has been positively evaluated and may be considered for inclusion in the roadmap in the future, depending on priorities and overall specific needs.
